From 41eb3bc894701034838a5dc126bf5492028d6b62 Mon Sep 17 00:00:00 2001
From: "River, Li" <river.li@intel.com>
Date: Thu, 24 Dec 2015 18:06:30 +0800
Subject: [PATCH 4/8] FOR_UPSTREAM [VPG]: drm/i915: Add ftrace for queue retire
 worker.

This commit adds ftrace event to track queue retire worker state.

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-27143
Change-Id: If5c99a65c3ad1eafe6f7c5c3562d57002adb6404
Signed-off-by: River, Li <river.li@intel.com>
Reviewed-on: https://android.intel.com:443/477657
Reviewed-on: https://android.intel.com:443/466811
---
 drivers/gpu/drm/i915/i915_gem.c   |  3 +++
 drivers/gpu/drm/i915/i915_trace.h | 21 +++++++++++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index f5ee914..05f8ac1 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -2744,6 +2744,9 @@ static void queue_retire_work(struct drm_i915_private *dev_priv,
 		 * scheduling more.
 		 */
 		dev_priv->mm.retire_work_timestamp = jiffies;
+		trace_queue_retire_work(time, true);
+	} else {
+		trace_queue_retire_work(time, false);
 	}
 }
 
diff --git a/drivers/gpu/drm/i915/i915_trace.h b/drivers/gpu/drm/i915/i915_trace.h
index 3582e5c..a1ab7be 100644
--- a/drivers/gpu/drm/i915/i915_trace.h
+++ b/drivers/gpu/drm/i915/i915_trace.h
@@ -1290,6 +1290,27 @@ TRACE_EVENT(execlists_context_queue,
 		__entry->ring, __entry->ctx, __entry->tail)
 );
 
+/* queue_retire_work - begin */
+TRACE_EVENT(queue_retire_work,
+	TP_PROTO(u64 time, bool bSuccess),
+
+	TP_ARGS(time, bSuccess),
+
+	TP_STRUCT__entry(
+			__field(u64, time)
+			__field(bool, bSuccess)
+	),
+
+	TP_fast_assign(
+			__entry->time = time;
+			__entry->bSuccess = bSuccess;
+	),
+
+	TP_printk("time=%lld, queue retire work sucess=%d",
+		__entry->time, __entry->bSuccess)
+);
+/* queue_retire_work - end */
+
 #endif /* _I915_TRACE_H_ */
 
 /* This part must be outside protection */
-- 
1.9.1

